### ISSUE TITLE
Solid.js integration

### ISSUE THREAD

**Issue by davedbase** [frandiox/vitedge#36]
Hi there, this looks like a great project. I'm investigating how much effort it would be to support Solid.js. Solid SSR is about 2x Marko according to tests against isomorphic-ui-benchmarks, so I think it make a fantastic candidate for Worker edge serving.

Is this as straightforward as replacing some Vite config options? If you provide some guidance I can likely handle a PR for you to entertain.

**Comment by frandiox** [comment]
@davedbase Hi, thanks a lot!
Is not *that* straightforward, but it might not be too complicated. I wrote more about it here (for Svelte): https://github.com/frandiox/vitedge/issues/10#issuecomment-808710180

The key is that we rely on the framework's router. I've no idea how the router works in Solid.js so not sure if it will be complicated 😅 

Ask me anything if you need more clarifications 👍

**Comment by davedbase** [comment]
@frandiox yup!

Ryan's working on a more consolidated version that will power our starter, Solid Start.

https://github.com/solidjs/solid-app-router currently supports SSR and is the closest thing to official. Our solid-hackernews example is already doing SSR with Vite and Cloudflare Worker: https://github.com/solidjs/solid-hackernews/tree/vite-ssr. So I think it's a matter of porting what Ryan has done over.

I know work is actively being done on the SSR API right now, but that shouldn't prohibit getting something going with vitedge.

PS. It would be *awesome* to get this working in Vitedge. Solid's SSR perf is...insane. Run that at the edge and you're flying. :-)

**Comment by frandiox** [comment]
@davedbase Hey! Are you aware of any news related to Solid or Solid router?

**Comment by davedbase** [comment]
@frandiox yeah! @ryansolid has been making great progress. I believe this is the branch that the work is being done on: https://github.com/solidjs/solid-app-router/tree/unified. Ryan's unifying different approaches into one.

**Comment by davedbase** [comment]
@frandiox so @ryansolid landed the new version of the router. Comes with a lot of bunch (https://twitter.com/webkoder/status/1417791942327508999?s=20). Would love to see what you guys do with it in vitedge. ;-)
